source/base/parameter_handler.cc

```diff
@@ -1329,285 +1328,269 @@ ParameterHandler::recursively_print_parameters(
   AssertThrow(out, ExcIO());
 
   // this function should not be necessary for XML or JSON output...
-  Assert((style != XML) && (style != JSON), ExcInternalError());
+  Assert(((style & XML) != XML) && ((style & JSON) != JSON),
```

Comment:
XML has to be written twice, JSON has to be written twice

Reply:
You could use `count_set_bits(style & (XML | JSON | Text)) == 1` if you had a function that counts the number of bits set in an integer. This way you make sure that exactly one bit is set.

An implementation of such a function is here: https://www.geeksforgeeks.org/count-set-bits-in-an-integer/

Reply:
Better choice is here: https://stackoverflow.com/questions/109023/how-to-count-the-number-of-set-bits-in-a-32-bit-integer

Reply:
Or, with C++20: https://en.cppreference.com/w/cpp/numeric/popcount